src/main/scala/zio/nio/channels/DatagramSocket.scala

```diff
@@ -0,0 +1,12 @@
+package zio.nio.channels
+
+import java.net.{ SocketException, DatagramSocket => JDatagramSocket }
+
+import zio.IO
+import zio.nio.InetSocketAddress
+
+final class DatagramSocket private[channels] (protected[channels] val socket: JDatagramSocket) {
```

Comment:
I'm not sure if we need this wrapper. The Java `DatagramSocket` isn't part of NIO, and the `bind` method is provided by the channel anyway. See also our `SocketChannel` wrapper just returns the Java `Socket` class from its `socket` method.

Reply:
Agree - removed.